server/src/interlockers/simple_sectional.bahn

```diff
@@ -0,0 +1,452 @@
+        # not been granted, the train ID would be empty
+        string conflict_train_id = get config route.train conflict_id
+        
+        # Route ID is not grantable if a conflicting route 
+        # has already been granted to a train
+        if conflict_train_id != ""
+            # Check whether route_id is grantable under the
+            # sectional route release strategy 
+            if !is_sectional_route_okay(route_id, conflict_id)
+                return false
+            end
+        end
+    end
+
+    return true
+end
+
+
+# For requested route requested_route_id and already granted route granted_route_id,
+# performs sectional interlocking check.
+# Returns true if requested_route_id is good-to-grant, otherwise returns false. 
+def is_sectional_route_okay(string requested_route_id, string granted_route_id) : bool
+    # 1. Determine the segments where the two routes intersect.
+    string requested_path[] = get config route.path requested_route_id
+    string granted_path[] = get config route.path granted_route_id
+    
+    for string requested_item in requested_path
+        # Check for matching segment IDs between the requested and granted paths
+        if (is requested_item a_segment)
+            for string granted_item in granted_path
+                if requested_item == granted_item
+                    # Matching segments -> critical section
+                    # Check whether any signals guarding entry into the critical section
+                    # are in aspect GO or SHUNT. If so, return false.
+                    if is_entry_signals_go_or_shunt_brute_force(requested_item)
+                        return false
+                    end
+                    
+                    # In the path of granted_route_id, check whether any segments between
+                    # the critical section and the signal before the critical section 
+                    # are occupied. If so, return false.
+                    if is_occupied_from_preceding_signal_to(granted_item, granted_route_id)
+                        return false
+                    end
+                end
+            end
+        end
+    end
+    
+    return true
+end
+
+
+# In the given route ID, checks whether any segment between the 
+# seg_id and the signal preceding seg_id is occupied. 
+# If so, returns false. Otherwise returns true.
+def is_occupied_from_preceding_signal_to(string seg_id, string route_id): bool
+    if route_id == "" || seg_id == ""
+        return false
+    end
+    # Determine preceding signal index and index of seg_id
+    string route_path[] = get config route.path route_id
+    int seg_id_occurrences_counter = 0
+    int seg_id_index_in_path = 0
+    int current_path_index = 0
+    int preceding_signal_index_in_path = 0
+    for string path_item in route_path0
+        if path_item == seg_id
+            seg_id_occurrences_counter = seg_id_occurrences_counter + 1
```

Comment:
Use `+=` so that the SCCharts compiler does not complain about the use of "old style" relative updates.